### PR TITLE
Add CI integration tests

### PR DESCRIPTION
**Pull request by valerifilipov**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    docker:
+      - image: aeternity/infrastructure:Split-terraform-modules-in-separate-repos-PT-165295661
+    working_directory: /infrastructure
+
+  check_terraform_changes: &check_terraform_changes
+    run:
+      name: Fail on Terraform changes
+      command: make check-terraform-changes
+
+  setup_secrets: &setup_secrets
```
Comment (dincho) [inline .circleci/config.yml]: This should be a command

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    steps:
+      - *setup_secrets
+      - run:
+          name: Fail on Terraform changes
+          command: |
+            make check-terraform-changes
+
+  # Checks the compatibility of latest package build and master bootstrap
+  integration_tests:
+    <<: *container_config
```
Comment (dincho) [inline .circleci/config.yml]: let's use custom executor

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    steps:
+      - *setup_secrets
+      - run:
+          name: Setup integration tests parameters
+          command: |
+            echo "export TF_VAR_bootstrap_version=v2.0.1" >> $BASH_ENV
```
Comment (dincho) [inline .circleci/config.yml]: I'm not sure about this version tho, why not `master` ?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      - integration_tests
+
+workflows:
+  check_deploy:
+    jobs:
```
Comment (dincho) [inline .circleci/config.yml]: you have to add `context:` to each job that needs secrets

Comment (valerifilipov) [reply .circleci/config.yml]: added in 9bd092f

File: .circleci/config.yml (hunk 3 above)
Comment (valerifilipov) [reply .circleci/config.yml]: removed this step, because defaults to master

File: test/test.tf [hunk 5]
```diff
@@ -0,0 +1,53 @@
+variable "vault_addr" {
```
Comment (lucafavatella) [inline test/test.tf]: I would expect to see PR re https://github.com/aeternity/infrastructure/blob/491acdd2057328bd83051cc43bf3e9f01799acac/.circleci/config.yml#L213 alongside this. It looks like duplicated file. Which one is canonical?

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          name: Integration environment cleanup
+          command: |
+            cd test && terraform destroy -parallelism=20 --auto-approve
+          when: always
+
+  setup_secrets:
+    steps:
+      - run:
+          name: Setup environment secrets
+          command: echo "source /infrastructure/import-secrets.sh" >> $BASH_ENV
```
Comment (lucafavatella) [inline .circleci/config.yml]: I see this repo terraform-aws-aenode-deploy used in several points in aeternity/infrastructure repo. Which repo depends on which? It seems chicken-and-egg to me.

**Comment by lucafavatella** [comment]
In https://circleci.com/gh/aeternity/terraform-aws-aenode-deploy/16 I read a couple of `Error reading secret/github/prod/token: Error making API request.`.

**Comment by valerifilipov** [comment]
> In https://circleci.com/gh/aeternity/terraform-aws-aenode-deploy/16 I read a couple of `Error reading secret/github/prod/token: Error making API request.`.

This is not related to this PR

File: test/health-check.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -Eeuo pipefail
+
+export AWS_DEFAULT_REGION=${AWS_DEFAULT_REGION:-ap-southeast-2}
+export AWS_DEFAULT_OUTPUT=${AWS_DEFAULT_OUTPUT:-text}
+
+IPS=$(aws ec2 describe-instances --filters "Name=tag:envid,Values=${ENVID:?}" --query "Reservations[*].Instances[*].PublicIpAddress" | tr '\t' '\n')
+test -n "$IPS" || echo "No instances found with envid: ${ENVID} in ${AWS_DEFAULT_REGION} region"
```
Comment (lucafavatella) [inline test/health-check.sh]: So if IPS is empty this script succeeds?

Comment (lucafavatella) [reply test/health-check.sh]: I would expect this checking number of nodes.

File: test/health-check.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -Eeuo pipefail
+
+export AWS_DEFAULT_REGION=${AWS_DEFAULT_REGION:-ap-southeast-2}
```
Comment (lucafavatella) [inline test/health-check.sh]: I find this default error prone. Though I do not have full picture of usage of these scripts.

Comment (dincho) [reply test/health-check.sh]: This is the region where we run the tests, it's more like hardcoded but I kept the option to be overwritten externally

File: test/health-check.sh (hunk 7 above)
Comment (dincho) [reply test/health-check.sh]: Good point, @valerifilipov do you mind improving it?

**Pull request merged**